app.py

```diff
@@ -59,79 +57,25 @@ def onboarding_message(event, client):
+    # Get the ids of the Slack channel, reaction, and timestamp of message reacted to associated with the incoming event
     channel_id = event.get("item", {}).get("channel")
-    user_id = event.get("user")
     reaction = event.get("reaction")
     message_ts = event.get("item", {}).get("ts")
-    message_thread_ts = event.get("item", {}).get("thread_ts")
 
-    # reactions and message only logged if the reaction is a round pushpin
+    # bot begins asking questions if the reaction is a round pushpin and then if the message has not been scraped already.
+    # if the message has been scraped, post a message reminding the user.
     if reaction == "round_pushpin":
-        # initializes the file for the messages
-        file = open("PythOnBoardingBot/messages.txt", "a+")
-
-        try:
-            # Call the conversations.history method using the WebClient
-            # The client passes the token you included in initialization    
-            result = client.conversations_history(
-                channel=channel_id,
-                inclusive=True,
-                oldest=message_ts,
-                latest=message_ts,
-                limit=1
-            )
-            message = result["messages"][0]
-
-            # Print message text
-            logger.info("{} has reacted with a '{}' to {}".format(user_id, reaction, message["text"]))
-            client.chat_postMessage(channel=channel_id, text="What do you want to title this wiki?", thread_ts=message_ts)
-
-            #  writes the message that was being reacted to into the messages.txt file
-            file.write("{}\n".format(message["text"]))
-            file.close()
-
-        except SlackApiError as e:
-            logger.info(f"Error: {e}")
-
-
-
-# =============== Pin Added Events ================ #
-# When a users pins a message the type of the event will be 'pin_added'.
-# Here we'll link the update_pin callback to the 'pin_added' event.
-@app.event("pin_added")
-def update_pin(event, client):
-    """Update the onboarding welcome message after receiving a "pin_added"
-    event from Slack. Update timestamp for welcome message as well.
-    """
-    # Get the ids of the Slack user and channel associated with the incoming event
-    channel_id = event.get("channel_id")
-    user_id = event.get("user")
-
-    # Get the original tutorial sent.
-    onboarding_tutorial = onboarding_tutorials_sent[channel_id][user_id]
-
-    # Mark the pin task as completed.
-    onboarding_tutorial.pin_task_completed = True
-
-    # Get the new message payload
-    message = onboarding_tutorial.get_message_payload()
-
-    # Post the updated message in Slack
-    updated_message = client.chat_update(**message)
+        if not message_in_documentation(message_ts):
+            client.chat_postMessage(channel=channel_id, text=onboarding_tutorial.question_one, thread_ts=message_ts)

```

Comment:
Yes -- that makes sense. I did consider this but wasn't too confident in changing it up since I thought it may have been a little too chaotic, but I should've definitely done that when I felt it too.

the onboarding_tutorial.py file has been renamed to bot_messages.py.

Reply:
Thanks for updating!